functions.php

```diff
@@ -0,0 +1,175 @@
+}
+
+/**
+ * Возвращает модифицированную дату
+ *
+ * Примеры использования:
+ * getModDate('2021-08-19 22:21:05'); // Array ([titleDate] => 19.08.2021 22:21, [rel] => 6 дней назад)
+ *
+ * @param string $date Дата в абсолютном формате
+ * 
+ * @return array
+ */
+function getModDate(string $date): array {
+    $modDate = [];
+
+    $modDate['titleDate'] = date('d.m.Y H:i', strtotime($date));
+    $diff = strtotime('now') - strtotime($date);
+
+    switch (true) {
+        case ($diff < HOUR):
+            $num = ceil($diff / MINUTE);
+            $res = get_noun_plural_form($num, 'минута', 'минуты', 'минут');
+            $modDate['rel'] = $num . ' ' . $res . ' назад';
+            break;
+        case ($diff >= HOUR  && $diff < DAY):
+            $num = ceil($diff / HOUR);
+            $res = get_noun_plural_form($num, 'час', 'часа', 'часов');
+            $modDate['rel'] = $num . ' ' . $res . ' назад';
+            break;
+        case ($diff >= DAY && $diff < WEEK):
+            $num = ceil($diff / DAY);
+            $res = get_noun_plural_form($num, 'день', 'дня', 'дней');
+            $modDate['rel'] = $num . ' ' . $res . ' назад';
+            break;
+        case ($diff >= WEEK&& $diff < FIVEWEEKS):
+            $num = ceil($diff / WEEK);
+            $res = get_noun_plural_form($num, 'неделя', 'недели', 'недель');
+            $modDate['rel'] = $num . ' ' . $res . ' назад';
+            break;
+        case ($diff >= FIVEWEEKS):
+            $num = ceil($diff / FIVEWEEKS);
+            $res = get_noun_plural_form($num, 'месяц', 'месяца', 'месяцев');
+            $modDate['rel'] = $num . ' ' . $res . ' назад';
+            break;
+    }
+
+    return $modDate;
+}
+
+/**
+ * Устанавливает новое соединение с сервером MySQL
+ * 
+ * @return mysqli Object $con Обьект mysqli
+ */
+function masqliConnect(): object {
+    $con = mysqli_connect('localhost', 'root', '', 'readme');
+
+    if (!$con) {
+       die('Ошибка соединения с сервером MySQL: ' . mysqli_connect_error());
+    }
+
+    mysqli_set_charset($con, 'utf8');
+
+    return $con;
+}
+
+/**
+ * Возвращает таблицу типов постов из базы данных
+ *
+ * @param mysqli Object $con Обьект mysqli
+ * 
+ * @return array
+ */
+function getPostTypes(object $con): array {
+    $query_types = 'SELECT * FROM content_types';
+
+    $stmt = mysqli_prepare($con, $query_types);
+    mysqli_stmt_execute($stmt);
+    $res = mysqli_stmt_get_result($stmt);
+
+    if (!$res) {
+        die('Ошибка получения данных: ' . mysqli_error($con));
+    }
+
+    $types = mysqli_fetch_all($res, MYSQLI_ASSOC);
+
+    return $types;
+}
+
+/**
+ * Возвращает обьединенную таблицу постов, пользователей и типа постов из базы данных
+ *
+ * @param $type_id тип поста
+ * @param string $cardsOnPageAll максимальное количество постов выводимых на странице
+ * @param mysqli Object $con Обьект mysqli
+ * 
+ * @return array
+ */
+function getPosts($type_id, string $cardsOnPageAll, object $con): array {
```

Comment:
```suggestion
function getPosts(?int $type_id, bool $cardsOnPageAll, object $con): array {
```
?int это int|null
ну а cardsOnPageAll не понял – почему строка?